examples/3d/generate_custom_mesh.rs

```diff
@@ -0,0 +1,238 @@
+            [0.0, 0.45], [0.0, 0.2], [1.0, 0.2], [1.0, 0.45],
+            // Assigning the UV coords for the forward side.
+            [0.0, 0.45], [0.0, 0.2], [1.0, 0.2], [1.0, 0.45],
+        ],
+    );
+
+    // For meshes with flat shading, normlas are orthogonal (pointing out) from the direction of
+    // the surface.
+    // Normals are required for correct lighting calculations.
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_NORMAL,
+        vec![
+            // Normals for the top side (towards +y)
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            // Normals for the bottom side (towards -y)
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            // Normals for the right side (towards +x)
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            // Normals for the left side (towards -x)
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            // Normals for the back side (towards +z)
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            // Normals for the forward side (towards -z)
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
+        ],
+    );
+
+    // Create the triangles out of the 24 vertices we created.
+    // To construct a square, we need 2 triangles, therfore 12 traingles in total.
+    // To construct a triangle, we need the indices of its 3 defined vertices, adding them one
+    // by one, in a counter-clockwise order (relative to the position of the viewer, the order
+    // should appear counter-clockwise from the front of the triangle). Read more about how to correctly build a mesh manually
+    // in the Bevy documentation of a Mesh, further examples and the implementation of the built-in shapes.
+    #[rustfmt::skip]
+    cube_mesh.set_indices(Some(Indices::U32(vec![
+        0,3,1 , 1,3,2, // triangles making up the top (+y) facing side.
+        4,5,7 , 5,6,7, // bottom (-y) 
+        8,11,9 , 9,11,10, // right (+x)
+        12,13,15 , 13,14,15, // left (-x)
+        16,19,17 , 17,19,18, // back (+z)
+        20,21,23 , 21,22,23, // forward (-z)
+    ])));
+
+    cube_mesh
+}
+
+// Function that changes the UV mapping of the mesh, to apply the other texture.
+fn toggle_texture(mesh_to_change: &mut Mesh) {
+    // Use .attributes_mut to iterate over all of the attributes of the mesh, and get a mutable
+    // reference to them.
+    for attribute in mesh_to_change.attributes_mut() {
+        // We only want to change the UV attributes.
+        if attribute.0 == Mesh::ATTRIBUTE_UV_0.id {
```

Comment:
Instead of iterating over all attributes and checking the ID. It is better to use the `mesh.attribute_mut` method as follow:

```rust
let attributes = mesh_to_change.attribute_mut(Mesh::ATTRIBUTE_UV_0).unwrap();
let VertexAttributeVales::Float32x2(attributes) = attributes else {
    return;
};
for uv_coord in attributes.iter_mut() {
    // etc.
```

Bevy examples should demonstrate the best way of doing something, so this is necessary.